### PR TITLE
Removed unnecessary origin keywords

### PR DESCRIPTION
**Pull request by gautamsagar99**
## PR summary

Removed origin keyword as it is not required in the provided examples.
https://matplotlib.org/devdocs/gallery/images_contours_and_fields/contourf_demo.html

### CODE REVIEW

**Comment by oscargus** [comment]
It does not produce identical results. Compare the second plot where the "innermost" part of the top-right contour is no longer yellow.

**Comment by rcomer** [comment]
@oscargus which versions are you comparing?  There seems to have been a bug in recent Matplotlib versions (but not main or the release candidate) that caused that yellow not to appear.  I already confused myself with that a couple if days ago: https://discourse.matplotlib.org/t/cmap-set-over-and-set-under-not-working-in-newer-python-environments/23999

I also wondered if it worth adding an example to show use of _origin_, and I see @gautamsagar99 asked the same on the issue (#26600).  I know there is also a tension with “how much is too much to put in an example”.

**Comment by oscargus** [comment]
Ahh, sorry I mixed up the versions (checked on my phone)...

**Comment by gautamsagar99** [comment]
Hey, 

To clarify I am using matplotlib - 3.7.1 version and I used colab for testing all the examples.
 
Here is the link for colab:

https://colab.research.google.com/drive/1XnsYLa8Tr0zuLe4hRDRu4KBJcJKlEnXX?usp=sharing

If you want me to add a new section with the below example I can add it.

  import matplotlib.pyplot as plt
  import numpy as np
  
  x = np.arange(1, 10)
  y = x.reshape(-1, 1)
  h = x * y
  
  cs = plt.contourf(h, levels=[10, 30, 50],
      colors=['#808080', '#A0A0A0', '#C0C0C0'], extend='both', origin="upper")
  cs.cmap.set_over('red')
  cs.cmap.set_under('blue')
  cs.changed()

**Comment by gautamsagar99** [comment]
@rcomer Let me add a new section below which clearly populates the example using origin keyword.

@story645 Yes we can add side by side which will be clearly differentiating for lower and upper origin value.

Let me Just add it and push it.

Thanks

**Comment by gautamsagar99** [comment]
Finally I want to add The below section to the documentation:

"**Customizing Contour Plots with the origin Keyword: **

This section demonstrates how to use the `origin` parameter in Matplotlib's `contourf` function to adjust the positioning of grid data. The `origin` parameter specifies whether data coordinates originate from the upper left ('upper') or lower left ('lower') corner of the grid. 

```
import matplotlib.pyplot as plt
import numpy as np

x = np.arange(1, 10)
y = x.reshape(-1, 1)
h = x * y

fig, (ax1, ax2) = plt.subplots(ncols=2)

for (ax, origin) in [(ax1, 'upper'), (ax2, 'lower')]:
    ax.set_title(f"origin={origin}")
    cs = ax.contourf(h, levels=[10, 30, 50],
                     colors=['#808080', '#A0A0A0', '#C0C0C0'],
                     extend='both', origin=origin)
    cs.cmap.set_over('red')
    cs.cmap.set_under('blue')
    cs.changed()

plt.show()
```

This code creates two side-by-side contour plots with different `origin` settings ('upper' and 'lower'). The `contourf` function utilizes the `origin` parameter to control how grid data coordinates are interpreted for plotting. The resulting visualizations highlight the effect of changing the origin on the contour plot's appearance."

**Comment by gautamsagar99** [comment]
@story645 I'm Confused regarding the images. Do we need to upload the image for the output(for the above example) or It is going to auto generate?

**Comment by gautamsagar99** [comment]
@story645 Thank you for your assistance! I've made the necessary changes and have also generated the documentation. Please review the updates and let me know if any further adjustments are required.

File: galleries/examples/images_contours_and_fields/contourf_demo.py [hunk 1]
```diff
@@ -104,13 +101,36 @@
...
+    cs = ax.contourf(X, Y, Z, levels, cmap=cmap, extend=extend)
     fig.colorbar(cs, ax=ax, shrink=0.9)
     ax.set_title("extend = %s" % extend)
     ax.locator_params(nbins=4)
 
 plt.show()
 
+# %%
+# Customizing Contour Plots with the origin Keyword
+# ------------------
```
[the remark this answers is not in the document]
Comment (gautamsagar99) [reply galleries/examples/images_contours_and_fields/contourf_demo.py:113]: I will Adjust this in small sentence like "Contour Plots using origin Keyword" and modify similarly based on the previous titles.

File: galleries/examples/images_contours_and_fields/contourf_demo.py [hunk 2]
```diff
@@ -80,8 +77,8 @@
 
 CS4 = ax2.contour(X, Y, Z, levels,
                   colors=('k',),
-                  linewidths=(3,),
-                  origin=origin)
+                  linewidths=(3,)
+                  )
```
[the remark this answers is not in the document]
Comment (gautamsagar99) [reply galleries/examples/images_contours_and_fields/contourf_demo.py:81]: I will try to wrap this in a single line or two

File: galleries/examples/images_contours_and_fields/contourf_demo.py [hunk 3]
```diff
@@ -104,13 +101,36 @@
...
     fig.colorbar(cs, ax=ax, shrink=0.9)
     ax.set_title("extend = %s" % extend)
     ax.locator_params(nbins=4)
 
 plt.show()
 
+# %%
+# Customizing Contour Plots with the origin Keyword
+# ------------------
+# This section demonstrates how to use the origin parameter in Contour
```
[the remark this answers is not in the document]
Comment (gautamsagar99) [reply galleries/examples/images_contours_and_fields/contourf_demo.py:114]: This one I will think of nice sentence which makes more valuable and more precise in explaining the below code snippet.

**Comment by gautamsagar99** [comment]
@story645 I have done all the changes I hope everything looks good. Let me know if there is anything if you might want me to modify.

File: galleries/examples/images_contours_and_fields/contourf_demo.py [hunk 4]
```diff
@@ -104,13 +96,36 @@
...
+fig, (ax1, ax2) = plt.subplots(ncols=2)
+# Note: lower and upper are the values provided
+# to origin argument in contourf.
+for (ax, origin) in [(ax1, 'upper'), (ax2, 'lower')]:
+    ax.set_title(f"origin={origin}")
+    cs = ax.contourf(h, levels=[10, 30, 50], colors=[
+                     '#808080', '#A0A0A0', '#C0C0C0'], extend='both', origin=origin)
+    cs.cmap.set_over('red')
+    cs.cmap.set_under('blue')
+    cs.changed()
```
Comment (timhoffm) [inline galleries/examples/images_contours_and_fields/contourf_demo.py:125]: You can simplify this to:

```suggestion
fig, (ax1, ax2) = plt.subplots(ncols=2)
ax1.set_title("origin='upper'")
ax2.set_title("origin='lower'")
cs = ax1.contourf(h, levels=np.arange(5, 70, 5), extend='both', origin="upper")
cs = ax2.contourf(h, levels=np.arange(5, 70, 5), extend='both', origin="lower")
```
grafik

Comment (gautamsagar99) [reply galleries/examples/images_contours_and_fields/contourf_demo.py:125]: This look more optimized. Thanks, Let me add it.

**Review by timhoffm — APPROVED** [review]

File: galleries/examples/images_contours_and_fields/contourf_demo.py [hunk 5]
```diff
@@ -104,13 +96,31 @@
...
     ax.set_title("extend = %s" % extend)
     ax.locator_params(nbins=4)
 
 plt.show()
 
+# %%
+# Customizing Contour Plots with the origin Keyword
+# -------------------------------------------------
+# This code showcases contour plot customization using the origin keyword,
+# demonstrating distinct upper and lower origin perspectives.
```
[the remark this answers is not in the document]
Comment (gautamsagar99) [reply galleries/examples/images_contours_and_fields/contourf_demo.py:110]: @story645 Thank you for sharing. I will update the changes and push it.

[the remark this answers is not in the document]
Comment (gautamsagar99) [reply galleries/examples/images_contours_and_fields/contourf_demo.py]: oh is it, Let me recommit and push the latest changes again.

File: galleries/examples/images_contours_and_fields/contourf_demo.py [hunk 6]
```diff
@@ -104,13 +96,31 @@
...
+    cs = ax.contourf(X, Y, Z, levels, cmap=cmap, extend=extend)
     fig.colorbar(cs, ax=ax, shrink=0.9)
     ax.set_title("extend = %s" % extend)
     ax.locator_params(nbins=4)
 
 plt.show()
 
+# %%
+# Orient contour plots using the origin keyword
+# -----------------------------------------------
```
Comment (QuLogic) [inline galleries/examples/images_contours_and_fields/contourf_demo.py:108]: Should match:
```suggestion
# Orient contour plots using the origin keyword
# ---------------------------------------------
```

**Comment by gautamsagar99** [comment]
Huge thanks to @story645  and @timhoffm 

I wanted to express my sincere appreciation for your tremendous support during my first open source contribution. Your guidance in understanding automatic documentations with proper formatting and your help in fixing my mistakes were incredibly valuable. Thanks for making this journey so enriching!

**Comment by gautamsagar99** [comment]
Sure, Let me make that small change and push it.

**Comment by gautamsagar99** [comment]
Thank you so much for your guidance throughout the process! Contributing to this open-source project has been an amazing learning experience, and I'm really grateful for the opportunity. I completely understand that reviewing and merging PRs is no small task, and your support in guiding me through it means a lot. Looking forward to continuing this journey and contributing more in the future! 🙌😊

**Pull request merged**